### DISCUSSION TITLE
Issues writing saves to GBA Pokemon carts

### DISCUSSION THREAD

**Discussion by hosser** [sanni/cartreader#405] [Need help]
Hi. All of a sudden I am unable to write saves to my Pokemon Ruby and Fire Red cartridges. I've used this exact device to write saves to them before so I've no idea what has changed. I have two Cart Readers - one older unit built and bought from Sanni via the Arduino forums and another built by the recent Save the Hero kickstarter project - both now give me the same error "Flashrom Type not supported ID: 6213" as soon as I select the "Write save" menu option. The cart info and flash type are detected correctly (Name: POKEMONFIRE, Cart ID: BPRE, Rom Size: 16 MB, Save: 1024 KB Flash, Checksum: 67, Version 1.1).

Any ideas? The cartridges are genuine, not repros. Interestingly I have two Leaf Green cartridges and these two cartridges allow me to enter the Write save menu successfully. I haven't actually tested writing a save to them as I don't want the same thing to happen to them. 

Is it possible that the save file I've written to the Ruby and Fire Red carts is causing this issue? Would erasing the save file allow me to write a save with the Cart Reader again?

Thanks for any help/advice.

**Comment by Dakkaron** [comment]
Did you do an update on the devices or did you leave them as they were?

Since you tested two devices with two cartridges, I'd guess, it shouldn't be a simple hardware problem (e.g. dirt). Anyway, could you, just to be sure, clean both cartriges' contacts thoroughly using contact spray?

I have one guess, and that would be bad. Could it be, that you accidentally set your cart reader to the wrong voltage? This might have fried the chip containing the save data.

To check that, can you try creating a new save file and see if it survives power loss? For that, save the file, turn of the gameboy and remove the cartridge for a few minutes. Then check if the save still exists.

If it doesn't, the cartridges are probably broken...

**Reply by hosser** [reply]
Thanks for your reply. Wrong voltage is possible, though I would have thought that would be more fatal - I can load and save the existing save data on the cartridge with no issue. I've also just erased the save file on there by holding Up + Select + B and selecting erase all save data, then started a new game, saved, turned off the GBA, removed the cartridge and checked it again after 5 minutes and the save file loads just fine. Then I put the cartridge in the Cart Reader and while I can read the save, if I select Write save I get the same 6213 error.

**Comment by Eless0** [comment]
Any update on this? im having the same issue with my copy of sapphire. Used to be able to read and write without issue, suddenly it won't write anymore but the game itself still saves just fine. mine wasn't even caused by writing a particular save, I just left the game unplayed for about a month and when I came back and needed to edit the save it wouldn't work.

**Reply by ferox1183** [reply]
Same problem with my copy of ruby. My emerald works fine. I opened both cardriges and the flash chips are differnt. Emerald works finde. Mabye the type of flash is actually not compatible. I tested another Card reader (Burnmaster) and its the same Issue. The Cardrigte Test Mode says unknown flash type. I dont have GBxCart to test.

**Reply by hosser** [reply]
I have been using a GBxCart without issue with this same cartridge that was causing me trouble on my OSCR. Haven't tried the OSCR again ever since I got the GBxCart.

**Comment by hosser** [comment]
I'm using a GBxCart RW and FlashGBX now, works fine.
